ShellScript/Bash.sublime-syntax

```diff
@@ -925,537 +824,732 @@ contexts:
     - match: \!
       scope: punctuation.definition.history.shell
 
+  cmd-builtin:
+    # cmd modifier commands are directly followed by simple commands
+    - match: '{{modifier_cmds}}'
+      scope: meta.function-call.identifier.shell meta.path.shell support.function.shell
+    # shell built-in commands with arbritary arguments
+    - match: '{{builtin_cmds}}'
+      scope: meta.function-call.identifier.shell meta.path.shell support.function.shell
+      push:
+        - cmd-args-meta
+        - cmd-args
+
   cmd-basic:
-    # basic user defined or external commands
+    - match: \$?\"
+      scope: punctuation.definition.quoted.begin.shell
+      set:
+        - cmd-name-body
+        - path-pattern-double-quoted-body
+        - path-pattern-quoted-begin
+    - match: \$\'
+      scope: punctuation.definition.quoted.begin.shell
```

Comment:
Found this doing some tests on one of my files. Should be punctuation.definition.string.begin, no? Also elsewhere in the file. My poor phone is not a good place for reviewing huge changes.

Reply:
Basically a shell only does quotation handling or quote removal. A quotation mark can appear at any position in a word. Thus - ignoring some escaping and expansions - `./"foo"/'bar'` is equal to `"./foo/bar"` both resolved to `./foo/bar`. You can even write `./fo"o/ba"r`.

Thus it actually was odd to scope some parts of such a word `string.quoted` while leaving others blank.

The current syntax therefore distinguishes `literals` and `strings`. While former is used to handle variables (`variable.function`, `variable.other`), the latter one is used for common command arguments.

`punctuation.definition.quoted` is used for quotation within variables, while `punctuation.definition.string` is for string scoped arguments.

On line 854 we are quoting a command, which is scoped `variable.function` but not string.

Reply:
I've choosen `string` scopes as it clearly distinguishes variables from normal arguments in most color schemes.

Technical a concept of quoted or unquoted strings known from other languages does not exist. Hence it would also be correct to just scope everything `meta.word.[quoted|unquoted]` and remove all string-like scopes.